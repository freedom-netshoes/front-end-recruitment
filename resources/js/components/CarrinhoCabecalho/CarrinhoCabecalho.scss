.carrinho__cabecalho {
    position: relative;
    height: 140px;
    padding-top: 60px;
    margin: 0;
    .icone {
        margin-top: -15px;
    }
    .contador {
        position: absolute;
        top: 74px;
        right: 310px;
        background-color: #dfbd00;
        color: #000;
        font-size: 12px;
        font-weight: bold;
        border-radius: 10px;
        width: 20px;
        height: 20px;
        line-height: 20px;
    }
    .titulo {
        font-size: 24px;
        font-weight: bold;
        color: #fff;
        margin-left: 20px;
        line-height: 20px;
    }
}
