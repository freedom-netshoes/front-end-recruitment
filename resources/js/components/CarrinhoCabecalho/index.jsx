import React, {Component, PropTypes} from 'react';
import CSSModules from 'react-css-modules';
import styles from './CarrinhoCabecalho.scss';

@CSSModules(styles)
class CarrinhoCabecalho extends Component {
    render () {		
		const {quantidadeCarrinho} = this.props;
	
        return (        
             <div className="center-xs row" styleName="carrinho__cabecalho">
                <figure styleName="icone">
                    <img src={this.props.imgPath} alt={this.props.titulo} />
                </figure>
                <div styleName="contador">{quantidadeCarrinho}</div>
                <div styleName="titulo">{this.props.titulo}</div>
            </div>       
        )
    }      
}

CarrinhoCabecalho.defaultProps = {
    imgPath: 'img/bag.svg',
    titulo: 'SACOLA',
    contador: '0'
};
CarrinhoCabecalho.propTypes = {
    imgPath: PropTypes.string,
    titulo: PropTypes.string,
    contador: PropTypes.string
};    

export default CarrinhoCabecalho;
