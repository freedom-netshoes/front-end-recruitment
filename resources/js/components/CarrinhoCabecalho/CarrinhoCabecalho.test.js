'use strict';

import { shallow } from 'enzyme';
import React from 'react';
import CarrinhoCabecalho from './';

it('Should CarrinhoCabecalho match to snapshot without props', () => {
  let tree = shallow(<CarrinhoCabecalho />);
  expect(tree).toMatchSnapshot();
});

it('Should CarrinhoCabecalho match to snapshot with anonymous props', () => {
  let tree = shallow(<CarrinhoCabecalho imgPath='' titulo='' quantidadeCarrinho='' />);
  expect(tree).toMatchSnapshot();
});

it('Should CarrinhoCabecalho match to snapshot simulating props', () => {
  let tree = shallow(<CarrinhoCabecalho imgPath='img/bag.svg' titulo='SACOLA' quantidadeCarrinho="0" />);  
  expect(tree).toMatchSnapshot();
});