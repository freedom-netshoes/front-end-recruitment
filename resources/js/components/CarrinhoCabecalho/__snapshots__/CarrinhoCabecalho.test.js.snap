// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should CarrinhoCabecalho match to snapshot simulating props 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="center-xs row carrinho__cabecalho"
>
    <figure
        className="icone"
    >
        <img
            alt="SACOLA"
            src="img/bag.svg"
        />
    </figure>
    <div
        className="contador"
    >
        0
    </div>
    <div
        className="titulo"
    >
        SACOLA
    </div>
</div>,
  "nodes": Array [
    <div
      className="center-xs row carrinho__cabecalho"
>
      <figure
            className="icone"
      >
            <img
                  alt="SACOLA"
                  src="img/bag.svg"
            />
      </figure>
      <div
            className="contador"
      >
            0
      </div>
      <div
            className="titulo"
      >
            SACOLA
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <CarrinhoCabecalho
        imgPath="img/bag.svg"
        quantidadeCarrinho="0"
        titulo="SACOLA"
/>,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": WrappedComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "imgPath": "img/bag.svg",
          "quantidadeCarrinho": "0",
          "titulo": "SACOLA",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="center-xs row carrinho__cabecalho"
>
          <figure
                    className="icone"
          >
                    <img
                              alt="SACOLA"
                              src="img/bag.svg"
                    />
          </figure>
          <div
                    className="contador"
          >
                    0
          </div>
          <div
                    className="titulo"
          >
                    SACOLA
          </div>
</div>,
        "_debugID": 6,
        "_renderedOutput": <div
          className="center-xs row carrinho__cabecalho"
>
          <figure
                    className="icone"
          >
                    <img
                              alt="SACOLA"
                              src="img/bag.svg"
                    />
          </figure>
          <div
                    className="contador"
          >
                    0
          </div>
          <div
                    className="titulo"
          >
                    SACOLA
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CarrinhoCabecalho
    imgPath="img/bag.svg"
    quantidadeCarrinho="0"
    titulo="SACOLA"
/>,
}
`;

exports[`Should CarrinhoCabecalho match to snapshot with anonymous props 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="center-xs row carrinho__cabecalho"
>
    <figure
        className="icone"
    >
        <img
            alt=""
            src=""
        />
    </figure>
    <div
        className="contador"
    />
    <div
        className="titulo"
    />
</div>,
  "nodes": Array [
    <div
      className="center-xs row carrinho__cabecalho"
>
      <figure
            className="icone"
      >
            <img
                  alt=""
                  src=""
            />
      </figure>
      <div
            className="contador"
      />
      <div
            className="titulo"
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <CarrinhoCabecalho
        imgPath=""
        quantidadeCarrinho=""
        titulo=""
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": WrappedComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "imgPath": "",
          "quantidadeCarrinho": "",
          "titulo": "",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="center-xs row carrinho__cabecalho"
>
          <figure
                    className="icone"
          >
                    <img
                              alt=""
                              src=""
                    />
          </figure>
          <div
                    className="contador"
          />
          <div
                    className="titulo"
          />
</div>,
        "_debugID": 4,
        "_renderedOutput": <div
          className="center-xs row carrinho__cabecalho"
>
          <figure
                    className="icone"
          >
                    <img
                              alt=""
                              src=""
                    />
          </figure>
          <div
                    className="contador"
          />
          <div
                    className="titulo"
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CarrinhoCabecalho
    imgPath=""
    quantidadeCarrinho=""
    titulo=""
/>,
}
`;

exports[`Should CarrinhoCabecalho match to snapshot without props 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="center-xs row carrinho__cabecalho"
>
    <figure
        className="icone"
    >
        <img
            alt="SACOLA"
            src="img/bag.svg"
        />
    </figure>
    <div
        className="contador"
    />
    <div
        className="titulo"
    >
        SACOLA
    </div>
</div>,
  "nodes": Array [
    <div
      className="center-xs row carrinho__cabecalho"
>
      <figure
            className="icone"
      >
            <img
                  alt="SACOLA"
                  src="img/bag.svg"
            />
      </figure>
      <div
            className="contador"
      />
      <div
            className="titulo"
      >
            SACOLA
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <CarrinhoCabecalho
        imgPath="img/bag.svg"
        quantidadeCarrinho=""
        titulo="SACOLA"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": WrappedComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "imgPath": "img/bag.svg",
          "quantidadeCarrinho": "",
          "titulo": "SACOLA",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="center-xs row carrinho__cabecalho"
>
          <figure
                    className="icone"
          >
                    <img
                              alt="SACOLA"
                              src="img/bag.svg"
                    />
          </figure>
          <div
                    className="contador"
          />
          <div
                    className="titulo"
          >
                    SACOLA
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="center-xs row carrinho__cabecalho"
>
          <figure
                    className="icone"
          >
                    <img
                              alt="SACOLA"
                              src="img/bag.svg"
                    />
          </figure>
          <div
                    className="contador"
          />
          <div
                    className="titulo"
          >
                    SACOLA
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CarrinhoCabecalho
    imgPath="img/bag.svg"
    quantidadeCarrinho=""
    titulo="SACOLA"
/>,
}
`;
